geonode_mapstore_client/client/js/utils/AppRoutesUtils.js

```diff
@@ -71,6 +71,7 @@ export const GEOSTORY_ROUTES = [{
     shouldNotRequestResources: true
 }];
 
+export const CATALOG_PAGES_ROUTES = ['/all', '/maps', '/documents', '/geostories', '/dashboards', '/datasets'];
```

Comment:
@dsuren1 We should have the dynamic extraction of the pathname and check if a corresponding {param}Page configuration exists. This way we don't need to extend this array in the future when new pages could be added (e.g. for 3dtiles, etc.).
There are multiple ways of achieving this, but in general I'd test the existence of the config and do a fallback on the route of the catalog otherwise. 
How long do you think it will take to implement it?

Reply:
@giohappy 
>We should have the dynamic extraction of the pathname and check if a corresponding {param}Page configuration exists. 

We have the implementation already in place. 

But if we don't want to list the predefined routes supported, then we should define a path that is uniquely distinguishable  ex: `/resource/:page` so catalog page path will look like (i.e for maps) `/catalogue/#/resource/maps`. Kindly let me know if you are fine with this new path and I will make the changes accordingly. It shouldn't take more than a hour to amend it.